torch/utils/data/dataloader.py

```diff
@@ -41,20 +41,28 @@ def create_fetcher(kind, dataset, auto_collation, collate_fn, drop_last):
 class _InfiniteConstantSampler(Sampler):
     r"""Analogous to ``itertools.repeat(None, None)``.
     Used as sampler for :class:`~torch.utils.data.IterableDataset`.
+
+    Arguments:
+        data_source (Dataset): dataset to sample from
     """
 
-    def __init__(self):
-        super(_InfiniteConstantSampler, self).__init__(None)
+    def __init__(self, data_source):
+        self.data_source = data_source
```

Comment:
So, one unintended consequence of this, is that you will now keep `data_source` live in `_InfiniteConstantSampler`, where previously we could free it. Do you think this might matter?

Reply:
I don't think it will matter. `data_source` is the iterable dataset here, and will already be kept alive as long as the data loader is alive (as the `loader.dataset`). This sampler is also only alive when the data loader is (as the `loader.sampler`). So there should be no lifetime change. Also, for non iterable datasets, we have been keeping them alive in sampler in a similar way.